File: api/alunos-app/app/models/aula.py
# app/models/aula.py
from sqlalchemy import Column, Integer, Text, Date, ForeignKey
from app.database import Base


class Aula(Base):
    __tablename__ = "aula"

    id = Column(Integer, primary_key=True)
    data = Column(Date, nullable=False)
    anotacoes = Column(Text, nullable=False)
    comentarios = Column(Text)
    proxima_aula = Column(Text)
    aluno_id = Column(Integer, ForeignKey("alunos.id"), nullable=False)

File: frontend/aluno-front/src/app/models/aula.ts
import { Aluno } from './aluno';

export class Aula {
  constructor(
    public id: number,
    public dataAula: Date,
    public anotacoes: string,
    public comentarios?: string,
    public proximaAula?: Date,
    public aluno?: Aluno
  ) {}
}
